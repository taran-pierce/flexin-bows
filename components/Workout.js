import WorkoutTable from './WorkoutTable'
import ImageCard from './ImageCard'
import PropTypes from 'prop-types'
import Icon from './Icon'

const Workout = ( props ) => {
  const data = props.data
  const plans = props.data.workout

  return (
    <section className={`workout ${props.showDesc ? 'active' : ''}`}>
      <style jsx>{`
        h1 {
          background-color: #7c6f5e;
          margin: 0;
          color: #dfd2c8;
          text-shadow: -1px -1px rgba(0,0,0,.3);
          padding: .5rem;
        }

        .flex-item.image {
          text-align: center;
        }
        .flex-item.table {
          margin-bottom: 1rem;
        }

        .workout {
          padding: 1rem;
        }

        .workout .workout-content {
          background: rgba(255,255,255,.5);
          border: 1px solid rgba(0,0,0,.4);
          box-shadow: 0 2px 2px rgba(0,0,0,.3);
          transition: opacity .5s ease-in-out,
                      max-height .5s ease-in-out,
                      box-shadow .5s ease-in-out,
                      padding .5s ease-in-out,
                      visibility .5s ease-in-out;
          opacity: 0;
          padding: 0 1rem;
          margin: 0 0 1rem;
          max-height: 0;
          visibility: hidden;
        }

        .workout.active .workout-content {
          opacity: 1;
          padding: 1rem;
          max-height: 600px;
          visibility: visible;
        }

        .workout.active h1 {
          box-shadow: none;
        }

        @media( min-width: 992px ) {

          .flex {
            display: flex;
            flex-direction: row;
            flex-wrap: nowrap;
          }
        
          .flex-item {
            flex-grow: 1;
            transition: width .5s ease-in-out;
          }

          .flex-item.table {
            width: 0;
            margin-right: 1rem;
          }
          .flex-item.table.active {
            width: 60%;
          }

          .flex-item.image {
            width: 100%;
          }
                
          .flex.large-image .flex-item.table.active + .flex.large-image .flex-item.image {
            width: 56%;
          }
          .flex.large-image .flex-item.table.active {
            width: 44%;
          }

          .flex.large-image .flex-item.table.active + .flex.small-image .flex-item.image {
            width: 40%;
          }
          .flex.small-image .flex-item.table.active {
            width: 60%;
          }
        }
      `}</style>
      <h1>{data.title}
        <Icon 
          name='angle-double-down'
          clickEvent={props.toggleDesc}
          active={props.showDesc}
        />
        <Icon 
          name='table'
          clickEvent={props.onTableSizeChange}
          active={props.tableSize === 'large' ? 1 : 0}
        />
      </h1>
      <div className={'workout-content'}>
        <p><strong>Frequency: </strong>{data.frequency}</p>
        <p><strong>Time: </strong>{data.time}</p>
        <p>{data.content}</p>
      </div>
      <div className={`flex ${props.imageSize === 'large' ? 'large-image' : 'small-image'}`}>
        <div className={ `flex-item table ${props.tableSize === 'large' ? 'active' :''}`}>
          <WorkoutTable
            plans={plans}
            activeTab={props.activeTab}
            onWorkoutClick={props.onWorkoutClick}
            onBodyPartClick={props.onBodyPartClick}
            tableSize={props.tableSize}
          />
        </div>
        <div className={'flex-item image'}>
          <ImageCard image={props.image} />
        </div>
      </div>
    </section>
  )
}

Workout.propTypes = {
  data: PropTypes.any.isRequired,
  activeTab: PropTypes.string,
  showDesc: PropTypes.number,
  imageSize: PropTypes.string,
  tableSize: PropTypes.string,
  toggleDesc: PropTypes.number,
  onSizeChange: PropTypes.func,
  onTableSizeChange: PropTypes.func,
  onBodyPartClick: PropTypes.func,
  onWorkoutClick: PropTypes.func,
  image: PropTypes.string,
}

export default Workout