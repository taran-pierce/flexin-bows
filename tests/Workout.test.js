import React from 'react';
import Workout from '../components/Workout';
import renderer from 'react-test-renderer';
import data from '../data/programs/better-body.json'

test('Does Workout render correctly?', () => {
  const component = renderer.create(
    <Workout data={data} />,
  );
  
  // verify component compiles correctly
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
