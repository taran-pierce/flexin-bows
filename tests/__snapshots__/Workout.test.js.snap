// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does Workout render correctly? 1`] = `
<section
  className="jsx-748357250 workout "
>
  <h1
    className="jsx-748357250"
  >
    20 Minute Better Body Workout
    <span
      className="jsx-1190048806 close"
      style={
        Object {
          "float": "right",
        }
      }
    >
      <i
        aria-hidden={true}
        className="fa fa-angle-double-down"
        style={
          Object {
            "color": "#666",
            "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
            "transition": "all .5s ease-in-out",
          }
        }
      />
    </span>
    <span
      className="jsx-1190048806 close"
      style={
        Object {
          "float": "right",
        }
      }
    >
      <i
        aria-hidden={true}
        className="fa fa-table"
        style={
          Object {
            "color": "#666",
            "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
            "transition": "all .5s ease-in-out",
          }
        }
      />
    </span>
  </h1>
  <div
    className="jsx-748357250 workout-content"
  >
    <p
      className="jsx-748357250"
    >
      <strong
        className="jsx-748357250"
      >
        Frequency: 
      </strong>
      3 Days per week (M-W-F)
    </p>
    <p
      className="jsx-748357250"
    >
      <strong
        className="jsx-748357250"
      >
        Time: 
      </strong>
      20 Minutes
    </p>
    <p
      className="jsx-748357250"
    >
      Start by performing one set of each exercise. Warm up with a light resistance that you can perform easily for 5/10 reps without fatigue. Focus on practicing and learning your technique before increasing the resistance. Then move to a more challenging resistance that you can perform no less than 10 reps and no more than 15 reps. As you become stronger, you can advance to two sets for each exercise. Complete all sets of each exercise before moving ont to the next one. Res 30 to 45 seconds between sets. Move slowly on each repo. Use a pace that would allow you to stop the movement instantly at any point in the repo. Count three seconds up and threes seconds down and work to fatigue during each set.
    </p>
  </div>
  <div
    className="jsx-748357250 flex small-image"
  >
    <div
      className="jsx-748357250 flex-item table "
    >
      <div
        className="jsx-2306742958 workout-table"
        style={
          Object {
            "height": "0",
            "opacity": 0,
            "transition": "all .5s ease-in-out",
            "visibility": "hidden",
          }
        }
      >
        <div
          className="jsx-1110752472"
        >
          <ul
            className="jsx-1110752472"
          >
            <li
              className="jsx-1110752472"
              style={
                Object {
                  "background": "rgba(255,255,255,.5)",
                }
              }
            >
              <a
                className="jsx-1110752472"
                data-id="better-body-M-W-F"
                href="#"
              >
                M-W-F
              </a>
            </li>
          </ul>
        </div>
        <div
          className="jsx-2306742958"
        >
          <table
            className="jsx-2306742958 hidden"
          >
            <thead
              className="jsx-2306742958"
            >
              <tr
                className="jsx-2306742958"
              >
                <th
                  className="jsx-2306742958"
                >
                  Body Part
                </th>
                <th
                  className="jsx-2306742958"
                >
                  Exercise
                </th>
                <th
                  className="jsx-2306742958"
                >
                  Sets
                </th>
                <th
                  className="jsx-2306742958"
                >
                  Reps
                </th>
              </tr>
            </thead>
            <tbody
              className="jsx-2306742958"
            >
              <tr
                className="jsx-2306742958"
              >
                <td
                  className="jsx-2306742958"
                >
                  Chest
                </td>
                <td
                  className="jsx-2306742958"
                >
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="bench-press-horizontal"
                      href="#"
                    >
                      Bench Press
                    </a>
                  </p>
                </td>
                <td
                  className="jsx-2306742958"
                >
                  1 - 2
                </td>
                <td
                  className="jsx-2306742958"
                >
                  10 - 15
                </td>
              </tr>
              <tr
                className="jsx-2306742958"
              >
                <td
                  className="jsx-2306742958"
                >
                  Back
                </td>
                <td
                  className="jsx-2306742958"
                >
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="shoulder-extension"
                      href="#"
                    >
                      Seated Lat Rows
                    </a>
                  </p>
                </td>
                <td
                  className="jsx-2306742958"
                >
                  1 - 2
                </td>
                <td
                  className="jsx-2306742958"
                >
                  10 - 15
                </td>
              </tr>
              <tr
                className="jsx-2306742958"
              >
                <td
                  className="jsx-2306742958"
                >
                  Shoulders
                </td>
                <td
                  className="jsx-2306742958"
                >
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="cross-over-rear-deltiod"
                      href="#"
                    >
                      Crossover Seated Rear Deltoid Rows
                    </a>
                  </p>
                </td>
                <td
                  className="jsx-2306742958"
                >
                  1 - 2
                </td>
                <td
                  className="jsx-2306742958"
                >
                  10 - 15
                </td>
              </tr>
              <tr
                className="jsx-2306742958"
              >
                <td
                  className="jsx-2306742958"
                >
                  Arms
                </td>
                <td
                  className="jsx-2306742958"
                >
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="bicep-curl"
                      href="#"
                    >
                      Standing Biceps Curl
                    </a>
                  </p>
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="triceps-pushdown"
                      href="#"
                    >
                      Triceps Pushdown
                    </a>
                  </p>
                </td>
                <td
                  className="jsx-2306742958"
                >
                  1 - 2
                </td>
                <td
                  className="jsx-2306742958"
                >
                  10 - 15
                </td>
              </tr>
              <tr
                className="jsx-2306742958"
              >
                <td
                  className="jsx-2306742958"
                >
                  Legs
                </td>
                <td
                  className="jsx-2306742958"
                >
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="calf-raise"
                      href="#"
                    >
                      Calf Raise
                    </a>
                  </p>
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="leg-press"
                      href="#"
                    >
                      Leg Press
                    </a>
                  </p>
                </td>
                <td
                  className="jsx-2306742958"
                >
                  1 - 2
                </td>
                <td
                  className="jsx-2306742958"
                >
                  10 - 15
                </td>
              </tr>
              <tr
                className="jsx-2306742958"
              >
                <td
                  className="jsx-2306742958"
                >
                  Trunk
                </td>
                <td
                  className="jsx-2306742958"
                >
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="trunk-rotation"
                      href="#"
                    >
                      Trunk Rotation
                    </a>
                  </p>
                  <p
                    className="jsx-2306742958"
                  >
                    <a
                      className="jsx-2306742958"
                      data-file="abdominal-crunch"
                      href="#"
                    >
                      Seated Abdominal Crunch
                    </a>
                  </p>
                </td>
                <td
                  className="jsx-2306742958"
                >
                  1 - 2
                </td>
                <td
                  className="jsx-2306742958"
                >
                  10 - 15
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      className="jsx-748357250 flex-item image"
    >
      <div
        className="jsx-776312227"
      >
        <img
          alt="get it done"
          className="jsx-776312227"
        />
      </div>
    </div>
  </div>
</section>
`;
